src/main.rs

```diff
@@ -250,26 +262,32 @@ fn move_camera(
             direction += Vec3::new(0., SPEED, 0.);
         }
 
+        direction = direction.normalize_or_zero();
+
         transform.translation += direction;
     }
 }
 
-fn detect_collisions(mut events: EventReader<CollisionEvent>, mut health_query: Query<&mut Health, With<Player>>,
-    spike_query: Query<&Spike>, player_query: Query<&Player>) {
+fn is_player(layers: CollisionLayers) -> bool {
+    layers.contains_group(Layer::Player) && !layers.contains_group(Layer::Enemy)
+}
+
+fn is_enemy(layers: CollisionLayers) -> bool {
+    !layers.contains_group(Layer::Player) && layers.contains_group(Layer::Enemy)
+}
+
+fn detect_collisions(mut events: EventReader<CollisionEvent>, mut health_query: Query<&mut Health, With<Player>>) {
 
-    for event in events.iter() {
+    for event in events.iter().filter(|e| e.is_started()) {    
         for mut health in health_query.iter_mut() {
```

Comment:
This will work with only 1 player, if someday you wanna have second player, this will be wrong, you should probably do something like 
```
let health = if let Some((health)) = health_query.iter_mut().next() { health } else { return; } 
```
This will again not work with multiple players, but will atleast be more correct in a way.